src/User.js

```diff
@@ -15,15 +15,24 @@ class User {
     }
 
     searchApprovedTrips(timeFrame) {
+        // this naming should probably change if it also calls this.searchPendingTrips()
```

Comment:
One solution could be have a function called `searchTrips` and with `searchApprovedTrips` and `searchPendingTrips`
It could look like this

```javascript
searchTrips(timeFrame) {
    const timeFrames = new Set(['present','upcoming','past','pending']);
    return timeFrames.has(timeFrame) ? this.searchApprovedTrips(timeFrame) : this.searchPendingTrips();
}
```